persistence-jdbc/core/src/main/scala/com/lightbend/lagom/internal/persistence/jdbc/SlickDbProvider.scala

```diff
@@ -68,7 +77,12 @@ private[lagom] object SlickDbProvider {
     )
   }
 
-  private def bindSlickDatabase(jndiDbName: String, slickDb: DatabaseDef, lifecycle: ApplicationLifecycle): Unit = {
+  private def bindSlickDatabase(
+      jndiDbName: String,
+      slickDb: DatabaseDef,
+      lifecycle: ApplicationLifecycle,
```

Comment:
Since these are `private[lagom]` APIs, why not to remove `ApplicationLifecycle` here and do JNDI unbinding on the same CS task where `slickDb` shutdown runs?

Reply:
I'm still using `applicationLifecycle` in https://github.com/lagom/lagom/pull/2426/files/f95deb47546dfe438a4e3dd2dd79bd10cb29dd62#diff-fe8d1d9cef036c8c764e7f025f0207dbR93

But I guess I could rewrite that into `CoordinatedShutdown` and cascade-drop `applicationLifecycle`.